draft-ietf-quic-transport.md

```diff
@@ -2675,201 +2676,289 @@ created, it MUST close the connection with error code STREAM_STATE_ERROR.
+which can allow for better stream prioritization.
 
-Due to reordering, an endpoint could continue receiving frames for the stream
-even after the stream is closed for sending.  Frames received after a peer
-closes a stream SHOULD be discarded.  An endpoint MAY choose to limit the period
-over which it ignores frames and treat frames that arrive after this time as
-being in error.
+In the "Send" state, an endpoint transmits - and retransmits as necessary - data
+in STREAM frames.  The endpoint respects the flow control limits of its peer,
+accepting MAX_STREAM_DATA frames.  It generates STREAM_BLOCKED frames if it
+encounters flow control limits.
 
-An endpoint may receive a RST_STREAM in this state, such as when the peer resets
-the stream after sending a FIN on it. In this case, the endpoint MAY discard any
-data that it already received on that stream. The endpoint SHOULD close the
-connection with a FINAL_OFFSET_ERROR if the received RST_STREAM carries a
-different offset from the one already established.
+After the application indicates that stream data is complete and a STREAM frame
+containing the FIN bit is sent, the send stream enters the "Data Sent" state.
+From this state, the endpoint only retransmits stream data as necessary.  The
+endpoint no longer needs to track flow control limits or send STREAM_BLOCKED
+frames for a send stream in this state.  The endpoint can ignore any
+MAX_STREAM_DATA frames it receives from its peer in this state; MAX_STREAM_DATA
+frames might be received until the peer receives the final stream offset.
 
-An endpoint will know the final offset of the data it receives on a stream when
-it reaches the "half-closed (remote)" state, see {{final-offset}} for details.
+Once all stream data has been successfully acknowledged, the send stream enters
+the "Data Recvd" state, which is a terminal state.
 
-A stream in this state can be used by the endpoint to send any frame that
-mentions a stream ID.  In this state, the endpoint MUST observe advertised
-stream and connection data limits (see {{flow-control}}).
+From any of the "Open", "Send", or "Data Sent" states, an application can signal
+that it wishes to abandon transmission of stream data.  Similarly, the endpoint
+might receive a STOP_SENDING frame from its peer.  This causes the endpoint to
+send a RST_STREAM frame, which causes the stream to enter the "Reset Sent"
+state.
 
-A stream transitions from this state to "closed" by completing transmission of
-all data.  This includes sending all data carried in STREAM frames including
-the terminal STREAM frame that contains a FIN flag.
+An endpoint MAY send a RST_STREAM to open a send stream; this causes the send
+stream to open and immediately transition to the "Reset Sent" state.
 
-A stream also becomes "closed" when the endpoint sends a RST_STREAM frame.
+Once a packet containing a RST_STREAM has been successfully acknowledged, the
```

Comment:
Nit:  From the receiver's side of the ACK, there are no unsuccessful acknowledgements.